motionplan/tpSpaceRRT_test.go

```diff
@@ -467,76 +446,128 @@ func TestPtgCheckPlan(t *testing.T) {
+			obstaclesExist:   true,
+			obsPosition:      r3.Vector{-2500, -40, 0},
+			obsDims:          r3.Vector{10, 10, 1},
+			observerFrame:    cameraFrame.Name(),
+			errorState:       r3.Vector{0, 0, 0},
+			startPosition:    r3.Vector{0, 0, 0},
+			startOrientation: spatialmath.NewOrientationVector().Degrees(),
+			planIndex:        0,
+			errorIsNil:       true,
+		},
+		{
+			name:             "obstacles NOT in world frame cause collision - integration test",
+			obstaclesExist:   true,
+			obsPosition:      r3.Vector{2500, 20, 0},
+			obsDims:          r3.Vector{10, 2000, 1},
+			observerFrame:    cameraFrame.Name(),
+			errorState:       r3.Vector{0, 0, 0},
+			startPosition:    r3.Vector{0, 0, 0},
+			startOrientation: spatialmath.NewOrientationVector().Degrees(),
+			planIndex:        0,
+			errorIsNil:       false,
+		},
+		{
+			name:             "checking from partial-plan, ensure success with obstacles - integration test",
+			obstaclesExist:   true,
+			obsPosition:      r3.Vector{0, 20, 0},
+			obsDims:          r3.Vector{10, 200, 1},
+			observerFrame:    referenceframe.World,
+			errorState:       r3.Vector{0, 0, 0},
+			startPosition:    r3.Vector{0, 0, 0},
+			startOrientation: ov,
+			planIndex:        2,
+			errorIsNil:       true,
+		},
+		{
+			name:             "verify partial plan with non-nil errorState and obstacle",
+			obstaclesExist:   true,
+			obsPosition:      r3.Vector{0, 0, 0},
+			obsDims:          r3.Vector{10, 10, 1},
+			observerFrame:    referenceframe.World,
+			errorState:       r3.Vector{0, 1000, 0},
+			startPosition:    r3.Vector{0, 1000, 0},
+			startOrientation: ov,
+			planIndex:        2,
+			errorIsNil:       true,
+		},
+	}
+
+	testFn := func(t *testing.T, tc testCase) {
+		t.Helper()
+		var worldState *referenceframe.WorldState
+		if tc.obstaclesExist {
+			position := spatialmath.NewPoseFromPoint(tc.obsPosition)
+			obstacle, err := spatialmath.NewBox(position, tc.obsDims, "box")
+			test.That(t, err, test.ShouldBeNil)
+
+			geoms := []spatialmath.Geometry{obstacle}
+			gifs := []*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(tc.observerFrame, geoms)}
+
+			worldState, err = referenceframe.NewWorldState(gifs, nil)
+			test.That(t, err, test.ShouldBeNil)
+		} else {
+			worldState = referenceframe.NewEmptyWorldState()
+		}
+
+		errorState := spatialmath.NewPoseFromPoint(tc.errorState)
+		inputs := referenceframe.FloatsToInputs([]float64{0, 0, 0})
+		startPose := spatialmath.NewPose(tc.startPosition, tc.startOrientation)
+
+		err := CheckPlan(ackermanFrame, steps[tc.planIndex:], worldState, fs, startPose, inputs, errorState, logger)
+		if tc.errorIsNil {
+			test.That(t, err, test.ShouldBeNil)
+		} else {
+			test.That(t, err, test.ShouldNotBeNil)
+		}
+	}
+
+	for _, tc := range testCases {
+		c := tc // needed to workaround loop variable not being captured by func literals
+		t.Run(c.name, func(t *testing.T) {
```

Comment:
Not a huge fan of reformatting tests this way.

Not only is it more lines of code, but it's much more difficult to debug when something does break.

Reply:
i will revert